odoo/addons/base/tests/test_configmanager.py

```diff
@@ -565,6 +570,132 @@ def test_06_cli(self):
             'limit_request': 100,
         })
 
+    def test_06_environ(self):
+        with file_open('base/tests/config/environ') as file:
+            os.environ.update({
+                x[0]: x[2].rstrip('\n')
+                for line in file.readlines()
+                if line and (x := line.partition('=')) and x[0]
+                and not x[0].startswith('#')
+            })
```

Comment:
The `if line` part is useless because there's always the `\n`.
The `partition('=')` makes line that lacks an `=` sign valid, better crash
Useless to call `readlines()` if we want to get rid the the `\n`: `read().splitlines()` is better.

```suggestion
            os.environ.update(dict([
                line.split('=', 1)
                for line in file.read().splitlines()
                if line and not line.startswith('#')
            ]))
```

Reply:
I don't want to crash on empty lines (you can have blocks in the file). I stripped the line return before calling `partition`.